### PR TITLE
S3

### PR DESCRIPTION
**Pull request by matthew-andrews**
Fewer moving parts, fewer dynos, faster deploy mechanism, more reliable local deploys……

@commuterjoy

### CODE REVIEW

File: deploy.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+				console.log("About to upload " + file);
+				return new Promise(function(resolve, reject) {
+					s3bucket.upload({
+						Key: 'assets/' + file,
+						ContentType: determineContentType(file),
+						ACL: 'public-read',
+						Body: content,
+
+						// Copying next-build-tools for now
+						CacheControl: 'public, max-age=604800000'
```
Comment (commuterjoy) [inline deploy.js]: Cloudfront is a bit of a pain to invalidate so tend towards lower cache times for non-hashed assets on s3. 

http://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/Invalidation.html#invalidating-objects-api

Comment (commuterjoy) [inline deploy.js]: Or at some point at an aws-purge to next-build-tools

**Comment by commuterjoy** [comment]
:+1: good, apart from my comments about s3 cache invalidation

File: deploy.js (hunk 1 above)
Comment (matthew-andrews) [inline deploy.js]: i was considering heavily encouraging people not to update existing assets…… only create new………

Comment (commuterjoy) [inline deploy.js]: Yup, but for favicon.ico and things in standard locations?

Comment (matthew-andrews) [inline deploy.js]: everything's under `/assets` namespace so nothing should be in a standard location……

Comment (matthew-andrews) [inline deploy.js]: oh grr there are special routes for `favicon.ico`.

**Pull request merged**